diff --git a/Lib/pty.py b/Lib/pty.py
index fefb63a..4cef056 100644
--- a/Lib/pty.py
+++ b/Lib/pty.py
@@ -184,7 +184,10 @@ def _copy(master_fd, master_read=_read, stdin_read=_read):
             i_buf = i_buf[n:]
 
         if stdin_avail and STDIN_FILENO in rfds:
-            data = stdin_read(STDIN_FILENO)
+            try:
+                data = stdin_read(STDIN_FILENO)
+            except OSError:
+                data = b""
             if not data:
                 stdin_avail = False
             else:
