diff --git a/plugin/controllers/models/services.py b/plugin/controllers/models/services.py
index b6ec7884..7237bdfe 100644
--- a/plugin/controllers/models/services.py
+++ b/plugin/controllers/models/services.py
@@ -41,7 +41,7 @@ except ImportError:
 # The fields fetched by filterName() and convertDesc() all need to be
 # html-escaped, so do it there.
 #
-from cgi import escape as html_escape
+from html import escape as html_escape
 
 def filterName(name, encode=True):
 	if name is not None:
