diff --git a/btdevicesmanager/CONTROL/control b/btdevicesmanager/CONTROL/control
index 5339e09d..2d4b29b9 100644
--- a/btdevicesmanager/CONTROL/control
+++ b/btdevicesmanager/CONTROL/control
@@ -2,4 +2,4 @@ Package: enigma2-plugin-extensions-btdevicesmanager
 Version: 1.0-r0
 Description: this is bt devices manger to pair e.x keyboard or mouse
 Maintainer: a4tech
-Depends: python3-pexpect, bluez5, btinit
+Depends:
