diff --git a/configure.ac b/configure.ac
index d3d08bb..b92fa99 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3604,7 +3604,6 @@ gst-libs/gst/player/Makefile
 gst-libs/gst/video/Makefile
 gst-libs/gst/audio/Makefile
 gst-libs/gst/opencv/Makefile
-gst-libs/gst/allocators/Makefile
 sys/Makefile
 sys/dshowdecwrapper/Makefile
 sys/acmenc/Makefile

diff --git a/gst-libs/gst/Makefile.am b/gst-libs/gst/Makefile.am
index 09efdfc..92130cd 100644
--- a/gst-libs/gst/Makefile.am
+++ b/gst-libs/gst/Makefile.am
@@ -15,12 +15,12 @@ OPENCV_DIR=opencv
 endif
 
 SUBDIRS = uridownloader adaptivedemux interfaces basecamerabinsrc codecparsers \
-	 insertbin mpegts base video audio player allocators isoff $(GL_DIR) $(WAYLAND_DIR) \
+	 insertbin mpegts base video audio player isoff $(GL_DIR) $(WAYLAND_DIR) \
 	 $(OPENCV_DIR)
 
 noinst_HEADERS = gst-i18n-plugin.h gettext.h glib-compat-private.h
 DIST_SUBDIRS = uridownloader adaptivedemux interfaces gl basecamerabinsrc \
-	codecparsers insertbin mpegts wayland opencv base video audio player allocators isoff
+	codecparsers insertbin mpegts wayland opencv base video audio player isoff
 
 #dependencies
 video, audio: base
@@ -29,7 +29,7 @@ adaptivedemux: uridownloader
 
 INDEPENDENT_SUBDIRS = \
 	interfaces basecamerabinsrc codecparsers insertbin uridownloader \
-	mpegts base player allocators isoff $(GL_DIR) $(WAYLAND_DIR) $(OPENCV_DIR)
+	mpegts base player isoff $(GL_DIR) $(WAYLAND_DIR) $(OPENCV_DIR)
 
 .PHONY: independent-subdirs $(INDEPENDENT_SUBDIRS)
 
