diff --git a/plugin/ui.py b/plugin/ui.py
index f7a3264..760a345 100644
--- a/plugin/ui.py
+++ b/plugin/ui.py
@@ -128,7 +128,7 @@ config.plugins.filecommander.path_left_selected = ConfigYesNo(default=True)
 config.plugins.filecommander.lastcursorposition = ConfigInteger(default=0)
 config.plugins.filecommander.showTaskCompleted_message = ConfigYesNo(default=True)
 config.plugins.filecommander.showScriptCompleted_message = ConfigYesNo(default=True)
-config.plugins.filecommander.hashes = ConfigSet(key_actions.hashes.keys(), default=["MD5"])
+config.plugins.filecommander.hashes = ConfigSet(list(key_actions.hashes.keys()), default=["MD5"])
 config.plugins.filecommander.bookmarks = ConfigLocations()
 config.plugins.filecommander.fake_entry = NoSave(ConfigNothing())
 
