diff --git a/lib/dvb/scan.cpp b/lib/dvb/scan.cpp
index 269ce4d..1f2dd01 100644
--- a/lib/dvb/scan.cpp
+++ b/lib/dvbs/can.cpp
@@ -307,8 +307,8 @@ RESULT eDVBScan::startFilter()
 				{
 					eDVBFrontendParametersCable parm;
 					m_ch_current->getDVBC(parm);
-					if ((tsid == 0x00d7 && abs(parm.frequency-618000) < 2000) ||
-						(tsid == 0x00d8 && abs(parm.frequency-626000) < 2000))
+					if ((tsid == 0x00d7 && fabs(parm.frequency-618000) < 2000) ||
+						(tsid == 0x00d8 && fabs(parm.frequency-626000) < 2000))
 						tsid = -1;
 				}
 			}
