diff --git a/lib/python/Plugins/SystemPlugins/Videomode/VideoHardware.py b/lib/python/Plugins/SystemPlugins/Videomode/VideoHardware.py
index bfab190ad..fae562cb3 100644
--- a/lib/python/Plugins/SystemPlugins/Videomode/VideoHardware.py
+++ b/lib/python/Plugins/SystemPlugins/Videomode/VideoHardware.py
@@ -54,9 +54,9 @@ class VideoHardware:
 								"auto": {50: "2160p25", 60: "2160p30", 24: "2160p24"}}
 
 	rates["2160p"] = {"50Hz": {50: "2160p50"},
-								"60Hz": {60: "2160p"},
-								"multi": {50: "2160p50", 60: "2160p"},
-								"auto": {50: "2160p50", 60: "2160p", 24: "2160p24"}}
+								"60Hz": {60: "2160p60"},
+								"multi": {50: "2160p50", 60: "2160p60"},
+								"auto": {50: "2160p50", 60: "2160p60", 24: "2160p24"}}
 
 	rates["PC"] = {
 		"1024x768": {60: "1024x768"}, # not possible on DM7025
