diff --git a/lib/python/Tools/StbHardware.py b/lib/python/Tools/StbHardware.py
index 2da266c..2c9beac 100644
--- a/lib/python/Tools/StbHardware.py
+++ b/lib/python/Tools/StbHardware.py
@@ -2,11 +2,15 @@ from os import path
 from fcntl import ioctl
 from struct import pack, unpack
 from time import time, localtime, gmtime
+from enigma import getBoxType
 
 def getFPVersion():
 	ret = None
 	try:
-		ret = long(open("/proc/stb/fp/version", "r").read())
+		if getBoxType() in ('dm7080','dm820','dm520','dm525','dm900','dm920'):
+			ret = open("/proc/stb/fp/version", "r").read()
+		else:
+			ret = long(open("/proc/stb/fp/version", "r").read())
 	except IOError:
 		try:
 			fp = open("/dev/dbox/fp0")
