diff --git a/lib/gdi/lcd.cpp b/lib/gdi/lcd.cpp
index 20e5f7824..bc64db0b9 100644
--- a/lib/gdi/lcd.cpp
+++ b/lib/gdi/lcd.cpp
@@ -265,7 +265,18 @@ void eDBoxLCD::update()
 			write(lcdfd, raw, _stride * height);
 		}
 		else
-			write(lcdfd, _buffer, _stride * res.height());
+		{
+			unsigned char gb_buffer[_stride * res.height()];
+			for (int offset = 0; offset < ((_stride * res.height())>>2); offset ++)
+			{
+				unsigned int src = 0;
+				if (offset%(_stride>>2) >= 4) // 4 is offset for dm9x0
+					src = ((unsigned int*)_buffer)[offset - 4];  
+			//                                             blue                         red                  green low                     green high
+			((unsigned int*)gb_buffer)[offset] = ((src >> 3) & 0x001F001F) | ((src << 3) & 0xF800F800) | ((src >> 8) & 0x00E000E0) | ((src << 8) & 0x07000700);
+			}
+			write(lcdfd, gb_buffer, _stride * res.height());
+		}
 	}
 	else /* lcd_type == 1 */
 	{
