diff --git a/lib/python/Screens/About.py b/lib/python/Screens/About.py
index 58b4b927a..c104e417a 100644
--- a/lib/python/Screens/About.py
+++ b/lib/python/Screens/About.py
@@ -25,7 +25,6 @@ class About(Screen):
 		hddsplit = skin.parameters.get("AboutHddSplit", 0)
 
 		AboutText = _("Hardware: ") + about.getHardwareTypeString() + "\n"
-		AboutText = _("Brand: ") + about.getHardwareBrand() + "\n"
 		AboutText += _("CPU: ") + about.getCPUInfoString() + "\n"
 		AboutText += _("CPU brand: ") + about.getCPUBrand() + "\n"
 		AboutText += _("Flash type: ") + about.getFlashType() + "\n"
