diff --git a/lib/python/Screens/SoftcamSetup.py  b/lib/python/Screens/SoftcamSetup.py 
index 0942eaf..0008e4f 100644
--- a/lib/python/Screens/SoftcamSetup.py 
+++ b/lib/python/Screens/SoftcamSetup.py 
@@ -92,7 +92,7 @@ class SoftcamSetup(Screen, ConfigListScreen):
 		if "oscam" in self.softcams.value.lower() and os.path.isfile('/usr/lib/enigma2/python/Plugins/Extensions/OscamStatus/plugin.py'):
 			from Plugins.Extensions.OscamStatus.plugin import OscamStatus
 			self.session.open(OscamStatus)
-		ppanelFileName = '/etc/ppanels/' + self.softcams.value + '.xml'
+			ppanelFileName = '/etc/ppanels/' + self.softcams.value + '.xml'
 		elif "cccam" in self.softcams.value.lower() and os.path.isfile('/usr/lib/enigma2/python/Plugins/Extensions/CCcamInfo/plugin.py'):
 			from Plugins.Extensions.CCcamInfo.plugin import CCcamInfoMain
 			self.session.open(CCcamInfoMain)
