diff --git a/lib/python/Screens/Hotkey.py b/lib/python/Screens/Hotkey.py
index f3a055057..3c2d0114a 100644
--- a/lib/python/Screens/Hotkey.py
+++ b/lib/python/Screens/Hotkey.py
@@ -670,7 +670,7 @@ class InfoBarHotkey:
 					return 0
 			elif selected[0] == "Module":
 				try:
-					exec("from %s import %s" % (selected[1], selected[2]))
+					exec("from %s import %s" % (selected[1], selected[2]), globals())
 					exec("self.session.open(%s)" % ",".join(selected[2:]))
 				except Exception as e:
 					print("[Hotkey] error during executing module %s, screen %s, %s" % (selected[1], selected[2], e))
diff --git a/lib/python/Screens/Menu.py b/lib/python/Screens/Menu.py
index da824d673..afa192a3a 100644
--- a/lib/python/Screens/Menu.py
+++ b/lib/python/Screens/Menu.py
@@ -65,7 +65,7 @@ class Menu(Screen, ProtectedScreen):
 		#	string (as we want to reference
 		#	stuff which is just imported)
 		if arg[0] != "":
-			exec("from %s import %s" % (arg[0], arg[1].split(",")[0]))
+			exec("from %s import %s" % (arg[0], arg[1].split(",")[0]), globals())
 			self.openDialog(*eval(arg[1]))
 
 	def nothing(self): #dummy
diff --git a/lib/python/Screens/Wizard.py b/lib/python/Screens/Wizard.py
index dcd408440..37da926bf 100644
--- a/lib/python/Screens/Wizard.py
+++ b/lib/python/Screens/Wizard.py
@@ -93,9 +93,9 @@ class Wizard(Screen):
 				self.wizard[self.lastStep]["config"]["type"] = type
 				if type == "ConfigList" or type == "standalone":
 					try:
-						exec("from Screens." + str(attrs.get('module')) + " import *")
+						exec("from Screens." + str(attrs.get('module')) + " import *", globals())
 					except:
-						exec("from " + str(attrs.get('module')) + " import *")
+						exec("from " + str(attrs.get('module')) + " import *", globals())
 
 					self.wizard[self.lastStep]["config"]["screen"] = eval(str(attrs.get('screen')))
 					if 'args' in attrs:
