diff --git a/lib/python/Screens/Setup.py b/lib/python/Screens/Setup.py
index 25d75c0a9..f82289201 100644
--- a/lib/python/Screens/Setup.py
+++ b/lib/python/Screens/Setup.py
@@ -88,8 +88,6 @@ class Setup(ConfigListScreen, Screen):
 		self["key_red"] = StaticText(_("Cancel"))
 		self["key_green"] = StaticText(_("OK"))
 		self["description"] = Label("")
-		self["HelpWindow"] = Pixmap()
-		self["HelpWindow"].hide()
 		self["VKeyIcon"] = Boolean(False)
 
 		self["actions"] = NumberActionMap(["SetupActions", "MenuActions"],
