diff --git a/lib/python/Screens/PictureInPicture.py b/lib/python/Screens/PictureInPicture.py
index 50ea6be..bdbdec2 100644
--- a/lib/python/Screens/PictureInPicture.py
+++ b/lib/python/Screens/PictureInPicture.py
@@ -117,7 +117,6 @@ class PictureInPicture(Screen):
 			y = 0
 		config.av.pip.save()
 		self.instance.move(ePoint(x, y))
-		self["video"].instance.move(ePoint(x, y))
 
 	def resize(self, w, h):
 		config.av.pip.value[2] = w
