diff --git a/lib/dvb_ci/dvbci_ccmgr.cpp b/lib/dvb_ci/dvbci_ccmgr.cpp
index b0e7663bb..9a02c62e6 100644
--- a/lib/dvb_ci/dvbci_ccmgr.cpp
+++ b/lib/dvb_ci/dvbci_ccmgr.cpp
@@ -980,7 +980,7 @@ bool eDVBCICcSession::ci_element_set_certificate(unsigned int id, X509 *cert)
 
 bool eDVBCICcSession::ci_element_set_hostid_from_certificate(unsigned int id, X509 *cert)
 {
-	X509_NAME *subject;
+	const X509_NAME *subject;
 	char hostid[16 + 1];
 	uint8_t bin_hostid[8];
 
