diff --git a/lib/python/Components/SystemInfo.py b/lib/python/Components/SystemInfo.py
index 01cba2fb8..1b633b5a1 100644
--- a/lib/python/Components/SystemInfo.py
+++ b/lib/python/Components/SystemInfo.py
@@ -97,7 +97,7 @@ SystemInfo["canMultiBoot"] = HardwareInfo().get_device_model() in ('hd51', 'h7',
 SystemInfo["canMode12"] = HardwareInfo().get_device_model() in ("hd51", "vs1500") and '192M' or HardwareInfo().get_device_model() in ("h7") and '200M'
 SystemInfo["CanDownmixDTSHD"] = fileExists("/proc/stb/audio/dtshd_choices") and fileCheck("/proc/stb/audio/dtshd")
 SystemInfo["CanDownmixWMApro"] = fileExists("/proc/stb/audio/wmapro_choices") and fileCheck("/proc/stb/audio/wmapro")
-SystemInfo["CanAC3plusTranscode"] = fileExists("/proc/stb/audio/ac3plus_choices") and fileCheck("/proc/stb/audio/ac3plus")
+SystemInfo["CanAC3plusTranscode"] = False
 SystemInfo["CanAACTranscode"] = fileExists("/proc/stb/audio/aac_transcode_choices") and fileCheck("/proc/stb/audio/aac_transcode")
 SystemInfo["HDRSupport"] = fileExists("/proc/stb/hdmi/hlg_support_choices") and fileCheck("/proc/stb/hdmi/hlg_support")
 SystemInfo["CanDownmixAC3"] = fileHas("/proc/stb/audio/ac3_choices", "downmix")
