diff --git a/lib/gdi/font.cpp b/lib/gdi/font.cpp
index 738b5c46e..c0cd21186 100644
--- a/lib/gdi/font.cpp
+++ b/lib/gdi/font.cpp
@@ -935,7 +935,7 @@ void eTextPara::blit(gDC &dc, const ePoint &offset, const gRGB &background, cons
 	gUnmanagedSurface *surface = target->surface;
 	gRGB currentforeground = foreground;
 
-	register int opcode = -1;
+	int opcode = -1;
 
 	__u32 lookup32_normal[16];
 	__u32 lookup32_invert[16];
@@ -1094,7 +1094,7 @@ void eTextPara::blit(gDC &dc, const ePoint &offset, const gRGB &background, cons
 			int rx = rxbase, ry = rybase;
 			__u8 *d = dbase;
 			__u8 *s = sbase;
-			register int sx = sxbase;
+			int sx = sxbase;
 			int sy = sybase;
 			if ((sy+ry) >= clip.rects[c].bottom())
 				sy = clip.rects[c].bottom()-ry;
@@ -1123,15 +1123,15 @@ void eTextPara::blit(gDC &dc, const ePoint &offset, const gRGB &background, cons
 				{
 				case 0: 		// 4bit lookup to 8bit
 					{
-					register int extra_buffer_stride = buffer_stride - sx;
-					register __u8 *td=d;
+					int extra_buffer_stride = buffer_stride - sx;
+					__u8 *td=d;
 					for (int ay = 0; ay < sy; ay++)
 					{
-						register int ax;
+						int ax;
 
 						for (ax=0; ax<sx; ax++)
 						{
-							register int b=(*s++)>>4;
+							int b=(*s++)>>4;
 							if(b)
 								*td=lookup8[b];
 							++td;
@@ -1143,14 +1143,14 @@ void eTextPara::blit(gDC &dc, const ePoint &offset, const gRGB &background, cons
 					break;
 				case 1:	// 8bit direct
 					{
-					register int extra_buffer_stride = buffer_stride - sx;
-					register __u8 *td=d;
+					int extra_buffer_stride = buffer_stride - sx;
+					__u8 *td=d;
 					for (int ay = 0; ay < sy; ay++)
 					{
-						register int ax;
+						int ax;
 						for (ax=0; ax<sx; ax++)
 						{
-							register int b=*s++;
+							int b=*s++;
 							*td++^=b;
 						}
 						s += extra_source_stride;
@@ -1161,13 +1161,13 @@ void eTextPara::blit(gDC &dc, const ePoint &offset, const gRGB &background, cons
 				case 2: // 16bit
                                         {
                                         int extra_buffer_stride = (buffer_stride >> 1) - sx;
-                                        register __u16 *td = (__u16*)d;
+                                        __u16 *td = (__u16*)d;
                                         for (int ay = 0; ay != sy; ay++)
                                         {
-                                                register int ax;
+                                                int ax;
                                                 for (ax = 0; ax != sx; ax++)
                                                 {
-                                                        register int b = (*s++) >> 4;
+                                                        int b = (*s++) >> 4;
 							if (b)
 								*td = lookup16[b];
                                                         ++td;
@@ -1179,14 +1179,14 @@ void eTextPara::blit(gDC &dc, const ePoint &offset, const gRGB &background, cons
                                         break;
 				case 3: // 32bit
 					{
-					register int extra_buffer_stride = (buffer_stride >> 2) - sx;
-					register __u32 *td=(__u32*)d;
+					int extra_buffer_stride = (buffer_stride >> 2) - sx;
+					__u32 *td=(__u32*)d;
 					for (int ay = 0; ay < sy; ay++)
 					{
-						register int ax;
+						int ax;
 						for (ax=0; ax<sx; ax++)
 						{
-							register int b=(*s++)>>4;
+							int b=(*s++)>>4;
 							if(b)
 								*td=lookup32[b];
 							++td;
