diff --git a/lib/python/Screens/SkinSelector.py b/lib/python/Screens/SkinSelector.py
index d5c76a9d1..4865479b2 100644
--- a/lib/python/Screens/SkinSelector.py
+++ b/lib/python/Screens/SkinSelector.py
@@ -87,7 +87,7 @@ class SkinSelector(Screen, HelpableScreen):
 			self["preview"].instance.setPixmap(ptr.__deref__())
 
 	def layoutFinished(self):
-		self.picload.setPara((self["preview"].instance.size().width(), self["preview"].instance.size().height(), 1.0, 1, 1, 1, "#ff000000"))
+		self.picload.setPara((self["preview"].instance.size().width(), self["preview"].instance.size().height(), 1, 1, 1, 1, "#ff000000"))
 		self.refreshList()
 
 	def refreshList(self):
