diff --git a/lib/dvb_ci/dvbci.cpp b/lib/dvb_ci/dvbci.cpp
index 165c749ef..c745fb718 100644
--- a/lib/dvb_ci/dvbci.cpp
+++ b/lib/dvb_ci/dvbci.cpp
@@ -749,7 +749,7 @@ void eDVBCIInterfaces::removePMTHandler(eDVBServicePMTHandler *pmthandler)
 
 				if (base_slot != slot)
 				{
-					eDVBCISlot *tmp = it->cislot;
+					eDVBCISlot *tmp = slot;
 					while(tmp->linked_next != slot)
 						tmp = tmp->linked_next;
 					ASSERT(tmp);
