diff --git a/lib/python/Components/About.py b/lib/python/Components/About.py
index 968e2ef84..935b8cb4a 100644
--- a/lib/python/Components/About.py
+++ b/lib/python/Components/About.py
@@ -151,10 +151,10 @@ def getDriverInstalledDate():
 	try:
 		from glob import glob
 		try:
-			driver = [x.split("-") for x in open(glob("/var/lib/opkg/info/*-dvb-modules-*.control")[0], "r") if x.startswith("Version:")][0]
+			driver = [x.split("-")[-2] for x in open(glob("/var/lib/opkg/info/*-dvb-modules-*.control")[0], "r") if x.startswith("Version:")][0]
 			if len(driver) == 2:
 				driver = driver[0].split('+')
-			return "%s-%s-%s" % (driver[1][:4], driver[1][4:6], driver[1][6:])
+			return "%s-%s-%s" % (driver[:4], driver[4:6], driver[6:])
 		except:
 			try:
 				driver = [x.split("Version:") for x in open(glob("/var/lib/opkg/info/*-dvb-proxy-*.control")[0], "r") if x.startswith("Version:")][0]
