--- a/src/plugin.py
+++ b/src/plugin.py
@@ -101,7 +101,7 @@
 	return TMDB_LANGUAGE_CODES.get(lang, 'rus')
 
 config.plugins.tmbd = ConfigSubsection()
-config.plugins.tmbd.locale = ConfigText(default="ru", fixed_size = False)
+config.plugins.tmbd.locale = ConfigText(default="en", fixed_size = False)
 config.plugins.tmbd.alrernative_locale = ConfigText(default="en", fixed_size = False)
 config.plugins.tmbd.available_languages = ConfigSelection(choices = [("1", _("Press OK"))], default = "1")
 config.plugins.tmbd.skins = ConfigSelection(default = "0", choices = [("0", _("Small poster")), ("1", _("Large poster"))])
