diff --git a/src/command/FileCommands.cxx b/src/command/FileCommands.cxx
index c27eaeb79..b59680ec0 100644
--- a/src/command/FileCommands.cxx
+++ b/src/command/FileCommands.cxx
@@ -132,7 +132,7 @@ public:
 
 	void OnPair(StringView key, StringView value) noexcept override {
 		if (IsValidName(key) && IsValidValue(value))
-			response.Fmt(FMT_STRING("{}: {}\n"), key, value);
+			response.Fmt(FMT_STRING("{}: {}\n"), key.data, value.data);
 	}
 };
 
diff --git a/src/TagPrint.cxx b/src/TagPrint.cxx
index e4d80e9bf..9c64ca30b 100644
--- a/src/TagPrint.cxx
+++ b/src/TagPrint.cxx
@@ -37,7 +37,7 @@ tag_print_types(Response &r) noexcept
 void
 tag_print(Response &r, TagType type, StringView value) noexcept
 {
-	r.Fmt(FMT_STRING("{}: {}\n"), tag_item_names[type], value);
+	r.Fmt(FMT_STRING("{}: {}\n"), tag_item_names[type], value.data);
 }
 
 void
