diff --git a/fs/unionfs/inode.c b/fs/unionfs/inode.c
index 7a6d747..3682693 100644
--- a/fs/unionfs/inode.c
+++ b/fs/unionfs/inode.c
@@ -900,7 +900,7 @@ static int unionfs_setattr(struct dentry *dentry, struct iattr *ia)
 	struct iattr lower_ia;
 
 	/* check if user has permission to change inode */
-	err = inode_change_ok(dentry->d_inode, ia);
+	err = setattr_prepare(dentry->d_inode, ia);
 	if (err)
 		goto out_err;
 
@@ -937,7 +937,7 @@ static int unionfs_setattr(struct dentry *dentry, struct iattr *ia)
 	lower_inode = lower_dentry->d_inode;
 
 	/* check if user has permission to change lower inode */
-	err = inode_change_ok(lower_inode, ia);
+	err = setattr_prepare(lower_inode, ia);
 	if (err)
 		goto out;
 
