diff --git a/plugin/OscamStatusSetup.py b/plugin/OscamStatusSetup.py
index 06a2266..9784f71 100644
--- a/plugin/OscamStatusSetup.py
+++ b/plugin/OscamStatusSetup.py
@@ -91,23 +91,18 @@ def parse_oscam_conf_file(filepath,data):
 		with open(filepath, 'r') as file_object:
 			line = file_object.readline()
 			while line:
-				# at each line check for a match with a regex
-				key, match = _parse_line(line)
-
-				if key == 'httpport':
-					port = match.group('httpport')
-					data.serverName="Autodetected"
+				if "httpport" in line.lower():
+					port = line.split("=")[1].strip()
+					data.serverName = "Autodetected"
 					if port[0] == '+':
 						data.useSSL = True
 						data.serverPort = port[1:]
 					else:
 						data.serverPort = port
-
-				if key == 'httpuser':
-					data.username = match.group('httpuser')
-
-				if key == 'httppwd':
-					data.password = match.group('httppwd')
+				if "httpuser" in line.lower():
+					data.username = line.split("=")[1].strip()
+				if "httppwd" in line.lower():
+					data.password = line.split("=")[1].strip()
 
 				line = file_object.readline()
 
