diff --git a/IPTVPlayer/components/iptvplayer.py b/IPTVPlayer/components/iptvplayer.py
index 27e7f4b0..9322acb9 100644
--- a/IPTVPlayer/components/iptvplayer.py
+++ b/IPTVPlayer/components/iptvplayer.py
@@ -11,7 +11,7 @@ except ImportError:
 from Screens.HelpMenu import HelpableScreen
 from Components.ActionMap import HelpableActionMap
 from Components.config import config
-from Components.AVSwitch import iAVSwitch
+from Plugins.SystemPlugins.Videomode.VideoHardware import video_hw as iAVSwitch
 from Screens.ChoiceBox import ChoiceBox
 from Components.ServiceEventTracker import ServiceEventTracker
 from enigma import iPlayableService, eTimer
