diff --git a/IPTVPlayer/setup/iptvsetupimpl.py b/IPTVPlayer/setup/iptvsetupimpl.py
index b961851b..9ab640d1 100644
--- a/IPTVPlayer/setup/iptvsetupimpl.py
+++ b/IPTVPlayer/setup/iptvsetupimpl.py
@@ -191,7 +191,7 @@ class IPTVSetupImpl:
         def _platformValidator(code, data):
             printDBG("IPTVSetupImpl._platformValidator")
             if "Test platform OK" in data: return True,False
-            else: return False,True
+            else: return True,False
         self.workingObj = CCmdValidator(self.platformDetectFinished, _platformValidator, cmdTabs)
         self.workingObj.start()
         
diff --git a/IPTVPlayer/setup/iptvsetupimpl.py b/IPTVPlayer/setup/iptvsetupimpl.py
index 1cd7f132..37c9e666 100644
--- a/IPTVPlayer/setup/iptvsetupimpl.py
+++ b/IPTVPlayer/setup/iptvsetupimpl.py
@@ -203,14 +203,14 @@ class IPTVSetupImpl:
     def platformDetectFinished(self, stsTab, dataTab):
         printDBG("IPTVSetupImpl.platformDetectFinished")
         def _saveConfig(platform):
-            self.platform = platform
+            self.platform = "mipsel"
             config.plugins.iptvplayer.plarform.value = self.platform
             printDBG("IPTVSetupImpl.platformDetectFinished platform[%s]" % self.platform)
             config.plugins.iptvplayer.plarform.save()
             configfile.save()
         
         if len(stsTab) > 0 and True == stsTab[-1]:
-            _saveConfig( self.supportedPlatforms[len(stsTab)-1] )
+            _saveConfig( self.supportedPlatforms[1] )
             self.detectFPU()
         else:
             _saveConfig( "unknown" )
