diff --git a/mytest.py b/mytest.py
index 836ddabcf..47c862bc9 100644
--- a/mytest.py
+++ b/mytest.py
@@ -23,12 +23,6 @@ enigma.eSocketNotifier = eBaseImpl.eSocketNotifier
 enigma.eConsoleAppContainer = eConsoleImpl.eConsoleAppContainer
 
 from Components.SystemInfo import SystemInfo
-if not SystemInfo["OpenVisionModule"]:
-	print("[mytest] Open Vision in multiboot! Now we have to remove what relies on our kernel module!")
-	from Components.Console import Console
-	Console = Console()
-	Console.ePopen('opkg remove enigma2-plugin-extensions-e2iplayer')
-	print("[mytest] Removed, this is on you not us!")
 
 from traceback import print_exc
 
