diff --git a/lib/python/Components/Network.py b/lib/python/Components/Network.py
index cfdd3e498..05202911e 100644
--- a/lib/python/Components/Network.py
+++ b/lib/python/Components/Network.py
@@ -6,7 +6,7 @@ from Components.PluginComponent import plugins
 from Plugins.Plugin import PluginDescriptor
 from Components.About import about
 
-oeversion = about.getVisionVersion()
+oeversion = "9.2"
 
 class Network:
 	def __init__(self):
