diff --git a/lib/python/Tools/HardwareInfo.py b/lib/python/Tools/HardwareInfo.py
index c2cdab076..0be9d12ff 100644
--- a/lib/python/Tools/HardwareInfo.py
+++ b/lib/python/Tools/HardwareInfo.py
@@ -3,7 +3,7 @@ hw_info = None
 class HardwareInfo:
 	device_name = _("unavailable")
 	device_model = None
-	device_brand = _("unavailable")
+	device_brand = _("Dreambox")
 	device_version = ""
 	device_revision = ""
 	device_hdmi = False
