diff --git a/skin.py b/skin.py
index 9c84bee9d..0628e04cd 100644
--- a/skin.py
+++ b/skin.py
@@ -82,7 +82,7 @@ def skin_user_skinname():
 # so the first screen found will be used.
 
 config.skin = ConfigSubsection()
-DEFAULT_SKIN = SystemInfo["HasFullHDSkinSupport"] and "OctEtFHD/skin.xml" or "PLi-HD/skin.xml"
+DEFAULT_SKIN = SystemInfo["HasFullHDSkinSupport"] and "PLi-FullNightHD/skin.xml" or "PLi-HD/skin.xml"
 # on SD hardware, HD skin will not be available
 if not fileExists(resolveFilename(SCOPE_SKIN, DEFAULT_SKIN)):
 	DEFAULT_SKIN = "skin.xml"
