diff --git a/lib/python/Tools/HardwareInfo.py b/lib/python/Tools/HardwareInfo.py
index d77f19cd2..8c8ed0b48 100644
--- a/lib/python/Tools/HardwareInfo.py
+++ b/lib/python/Tools/HardwareInfo.py
@@ -28,7 +28,7 @@ class HardwareInfo:
 
 		# Name ... bit odd, but history prevails
 		try:
-			self.device_name = open("/etc/model").read().strip()
+			self.device_name = open("/proc/stb/info/model").read().strip()
 		except:
 			pass
 
