diff --git a/lib/python/Screens/About.py b/lib/python/Screens/About.py
index e27ad3539..1ea6bb61c 100644
--- a/lib/python/Screens/About.py
+++ b/lib/python/Screens/About.py
@@ -34,7 +34,9 @@ class About(Screen):
 		AboutText += _("Open Vision revision: ") + about.getVisionRevision() + "\n"
 		AboutText += _("Open Vision module: ") + about.getVisionModule() + "\n"
 		AboutText += _("Build date: ") + about.getBuildDateString() + "\n"
-		AboutText += _("Last upgrade: ") + about.getUpdateDateString() + "\n"
+		ImageVersion = _("Last upgrade: ") + about.getImageVersionString()
+		self["ImageVersion"] = StaticText(ImageVersion)
+		AboutText += ImageVersion + "\n"
 
 		# [WanWizard] Removed until we find a reliable way to determine the installation date
 		# AboutText += _("Installed: ") + about.getFlashDateString() + "\n"
