diff --git a/plugin/backupdmm.sh b/plugin/backupdmm.sh
index c675098..26cb945 100755
--- a/plugin/backupdmm.sh
+++ b/plugin/backupdmm.sh
@@ -12,11 +12,7 @@ else
 	LIBDIR="/usr/lib"
 fi
 
-if [ -f /etc/visionversion ]; then
-	VISIONVERSION=`cat /etc/visionversion | sed "s/\..*//"`
-else
-	VISIONVERSION="7"
-fi
+VISIONVERSION="9"
 
 if [ $VISIONVERSION == "7" ]; then
 	LS1="-el"
diff --git a/plugin/backupsuite.sh b/plugin/backupsuite.sh
index 9eb85bd..a49dc0c 100755
--- a/plugin/backupsuite.sh
+++ b/plugin/backupsuite.sh
@@ -12,11 +12,7 @@ else
 	LIBDIR="/usr/lib"
 fi
 
-if [ -f /etc/visionversion ]; then
-	VISIONVERSION=`cat /etc/visionversion | sed "s/\..*//"`
-else
-	VISIONVERSION="7"
-fi
+VISIONVERSION="9"
 
 if [ $VISIONVERSION == "7" ]; then
 	LS1="-e1"
