diff --git a/plugin/backupdmm.sh b/plugin/backupdmm.sh
index 802f62b..96b497c 100755
--- a/plugin/backupdmm.sh
+++ b/plugin/backupdmm.sh
@@ -5,11 +5,7 @@
 #
 #!/bin/sh
 
-if [ -f /etc/visionversion ]; then
-	VISIONVERSION=`cat /etc/visionversion | sed "s/\..*//"`
-else
-	VISIONVERSION="7"
-fi
+VISIONVERSION="9"
 
 if [ $VISIONVERSION == "7" ]; then
 	LS1="-el"
diff --git a/plugin/backupsuite.sh b/plugin/backupsuite.sh
index b6da5f6..dcca8c8 100755
--- a/plugin/backupsuite.sh
+++ b/plugin/backupsuite.sh
@@ -5,11 +5,7 @@
 #
 #!/bin/sh
 
-if [ -f /etc/visionversion ]; then
-	VISIONVERSION=`cat /etc/visionversion | sed "s/\..*//"`
-else
-	VISIONVERSION="7"
-fi
+VISIONVERSION="9"
 
 if [ $VISIONVERSION == "7" ]; then
 	LS1="-e1"
