diff --git a/libusb/os/linux_usbfs.c b/libusb/os/linux_usbfs.c
index c300675..c8ecbb0 100644
--- a/libusb/os/linux_usbfs.c
+++ b/libusb/os/linux_usbfs.c
@@ -1429,7 +1429,7 @@ static int op_get_configuration(struct libusb_device_handle *handle,
 	uint8_t *config)
 {
 	struct linux_device_priv *priv = usbi_get_device_priv(handle->dev);
-	int active_config;
+	int active_config = 0;
 	int r;
 
 	if (priv->sysfs_dir) {
