diff --git a/plugin/controllers/models/movies.py b/plugin/controllers/models/movies.py
index 9e9fa78f..217aced2 100644
--- a/plugin/controllers/models/movies.py
+++ b/plugin/controllers/models/movies.py
@@ -31,7 +31,7 @@ from ServiceReference import ServiceReference
 from Components.config import config
 from Components.MovieList import MovieList
 from Screens.MovieSelection import defaultMoviePath
-from Tools.CopyFiles import copyFiles, moveFiles
+from Screens.CopyFiles import copyFiles, moveFiles
 from Tools.Directories import fileExists
 from ..i18n import _
 from ..utilities import getUrlArg2, toString
