diff --git a/lib/direct/os/waitqueue.h b/lib/direct/os/waitqueue.h
index cb45310..57853b3 100644
--- a/lib/direct/os/waitqueue.h
+++ b/lib/direct/os/waitqueue.h
@@ -37,7 +37,7 @@
 
 /**********************************************************************************************************************/
 
-#ifndef DIRECT_OS_WAITQUEUE_DEFINED
+#ifdef DIRECT_OS_WAITQUEUE_DEFINED
 DirectResult DIRECT_API direct_waitqueue_init        ( DirectWaitQueue *queue );
 
 DirectResult DIRECT_API direct_waitqueue_wait        ( DirectWaitQueue *queue, DirectMutex *mutex );
