diff --git a/gst-libs/gst/pbutils/gstdiscoverer.c b/gst-libs/gst/pbutils/gstdiscoverer.c
index 1bf5b8d6c..965297948 100644
--- a/gst-libs/gst/pbutils/gstdiscoverer.c
+++ b/gst-libs/gst/pbutils/gstdiscoverer.c
@@ -1797,7 +1797,7 @@ _serialized_info_get_path (GstDiscoverer * dc, gchar * uri)
     goto done;
   }
 
-  tmp = g_strdup_printf ("%s-%" G_GSIZE_FORMAT "-%" G_GINT64_FORMAT,
+  tmp = g_strdup_printf ("%s-%lli" G_GSIZE_FORMAT "-%li" G_GINT64_FORMAT,
       location, file_status.st_size, file_status.st_mtime);
   cs = g_checksum_new (G_CHECKSUM_SHA1);
   g_checksum_update (cs, (const guchar *) tmp, strlen (tmp));
