diff --git a/gst-libs/gst/video/gstvideodecoder.c b/gst-libs/gst/video/gstvideodecoder.c
index 6490179..7f8c4a5 100644
--- a/gst-libs/gst/video/gstvideodecoder.c
+++ b/gst-libs/gst/video/gstvideodecoder.c
@@ -3317,7 +3317,7 @@ gst_video_decoder_have_frame (GstVideoDecoder * decoder)
   guint flags;
   GstFlowReturn ret = GST_FLOW_OK;
 
-  GST_LOG_OBJECT (decoder, "have_frame at offset %" G_GSIZE_FORMAT,
+  GST_LOG_OBJECT (decoder, "have_frame at offset %llu" G_GSIZE_FORMAT,
       priv->frame_offset);
 
   GST_VIDEO_DECODER_STREAM_LOCK (decoder);
