diff --git a/plugin/backupdmm.sh b/plugin/backupdmm.sh
index 680a5e0..213a856 100755
--- a/plugin/backupdmm.sh
+++ b/plugin/backupdmm.sh
@@ -54,7 +54,7 @@ big_fail()
 if [ -d $WORKDIR ] ; then
 	log "FAIL!"
 	log "Content so far of the working directory $WORKDIR "
-	ls -el $WORKDIR >> $LOGFILE
+	ls -l $WORKDIR >> $LOGFILE
 fi
 clean_up
 echo $RED
@@ -104,7 +104,7 @@ backup_made()
 echo $LINE
 $SHOW "message10" ; echo "$MAINDEST" 	# USB Image created in:
 $SHOW "message23"		# "The content of the folder is:"
-ls "$MAINDEST" -e1rSh | sed 's/-.........    1//'
+ls "$MAINDEST" -1rSh | sed 's/-.........    1//'
 echo $LINE
 if  [ $HARDDISK != 1 ]; then
 	$SHOW "message11" ; echo "$EXTRA"		# and there is made an extra copy in:
@@ -119,7 +119,7 @@ backup_made_nfi()
 echo $LINE
 $SHOW "message42" ; echo "$MAINDEST" 	# NFI Image created in: 
 $SHOW "message23"		# "The content of the folder is:"
-ls "$MAINDEST" -e1rSh | sed 's/-.........    1//' 
+ls "$MAINDEST" -1rSh | sed 's/-.........    1//' 
 echo $LINE
 if  [ $HARDDISK != 1 ]; then
 	$SHOW "message11" ; echo "$EXTRA"		# and there is made an extra copy in:
@@ -367,8 +367,8 @@ echo -n $YELLOW
 {
 $SHOW "message24"  ; printf "%d.%02d " $MINUTES $SECONDS ; $SHOW "message25"
 } 2>&1 | tee -a $LOGFILE
-ROOTSIZE=`ls "$MAINDEST" -e1S | grep root | awk {'print $3'} ` 
-KERNELSIZE=`ls "$MAINDEST" -e1S | grep kernel | awk {'print $3'} ` 
+ROOTSIZE=`ls "$MAINDEST" -1S | grep root | awk {'print $3'} ` 
+KERNELSIZE=`ls "$MAINDEST" -1S | grep kernel | awk {'print $3'} ` 
 TOTALSIZE=$((($ROOTSIZE+$KERNELSIZE)/1024))
 SPEED=$(( $TOTALSIZE/$DIFF ))
 echo $SPEED > /usr/lib/enigma2/python/Plugins/Extensions/BackupSuite/speed.txt
@@ -513,8 +513,8 @@ echo -n $YELLOW
 {
 $SHOW "message24"  ; printf "%d.%02d " $MINUTES $SECONDS ; $SHOW "message25"
 } 2>&1 | tee -a $LOGFILE
-ROOTSIZE=`ls "$MAINDEST" -e1S | grep root | awk {'print $3'} `
-KERNELSIZE=`ls "$MAINDEST" -e1S | grep kernel | awk {'print $3'} `
+ROOTSIZE=`ls "$MAINDEST" -1S | grep root | awk {'print $3'} `
+KERNELSIZE=`ls "$MAINDEST" -1S | grep kernel | awk {'print $3'} `
 TOTALSIZE=$((($ROOTSIZE+$KERNELSIZE)/1024))
 SPEED=$(( $TOTALSIZE/$DIFF ))
 echo $SPEED > /usr/lib/enigma2/python/Plugins/Extensions/BackupSuite/speed.txt
diff --git a/plugin/backupsuite.sh b/plugin/backupsuite.sh
index 65f1789..29713a3 100755
--- a/plugin/backupsuite.sh
+++ b/plugin/backupsuite.sh
@@ -54,7 +54,7 @@ big_fail()
 if [ -d $WORKDIR ] ; then
 	log "FAIL!"
 	log "Content so far of the working directory $WORKDIR "
-	ls -el $WORKDIR >> $LOGFILE
+	ls -l $WORKDIR >> $LOGFILE
 fi
 clean_up
 echo $RED
@@ -104,7 +104,7 @@ backup_made()
 echo $LINE
 $SHOW "message10" ; echo "$MAINDEST" 	# USB Image created in:
 $SHOW "message23"		# "The content of the folder is:"
-ls "$MAINDEST" -e1rSh | sed 's/-.........    1//'
+ls "$MAINDEST" -1rSh | sed 's/-.........    1//'
 echo $LINE
 if  [ $HARDDISK != 1 ]; then
 	$SHOW "message11" ; echo "$EXTRA"		# and there is made an extra copy in:
@@ -317,7 +317,7 @@ if [ $ROOTNAME != "rootfs.tar.bz2" -o $SEARCH = "h9" -o $SEARCH = "h9combo" -o $
 	$NANDDUMP /dev/$MTDPLACE -qf "$WORKDIR/$KERNELNAME"
 	if [ -f "$WORKDIR/$KERNELNAME" ] ; then
 		echo -n "Kernel dumped  :"  >> $LOGFILE
-		ls -e1 "$WORKDIR/$KERNELNAME" | sed 's/-r.*   1//' >> $LOGFILE
+		ls -1 "$WORKDIR/$KERNELNAME" | sed 's/-r.*   1//' >> $LOGFILE
 	else
 		log "$WORKDIR/$KERNELNAME NOT FOUND"
 		big_fail
@@ -352,7 +352,7 @@ if [ $ROOTNAME != "rootfs.tar.bz2" ] ; then
 	$MKFS -r /tmp/bi/root -o "$WORKDIR/root.ubi" $MKUBIFS_ARGS
 	if [ -f "$WORKDIR/root.ubi" ] ; then
 		echo -n "ROOT.UBI MADE  :" >> $LOGFILE
-		ls -e1 "$WORKDIR/root.ubi" | sed 's/-r.*   1//' >> $LOGFILE
+		ls -1 "$WORKDIR/root.ubi" | sed 's/-r.*   1//' >> $LOGFILE
 		UBISIZE=`cat "$WORKDIR/root.ubi" | wc -c`
 		if [ "$UBISIZE" -eq 0 ] ; then
 			$SHOW "message39" 2>&1 | tee -a $LOGFILE
@@ -368,7 +368,7 @@ if [ $ROOTNAME != "rootfs.tar.bz2" ] ; then
 	chmod 644 "$WORKDIR/$ROOTNAME"
 	if [ -f "$WORKDIR/$ROOTNAME" ] ; then
 		echo -n "$ROOTNAME MADE:" >> $LOGFILE
-		ls -e1 "$WORKDIR/$ROOTNAME" | sed 's/-r.*   1//' >> $LOGFILE
+		ls -1 "$WORKDIR/$ROOTNAME" | sed 's/-r.*   1//' >> $LOGFILE
 	else
 		echo "$WORKDIR/$ROOTNAME NOT FOUND"  >> $LOGFILE
 		big_fail
@@ -453,8 +453,8 @@ echo -n $YELLOW
 {
 $SHOW "message24"  ; printf "%d.%02d " $MINUTES $SECONDS ; $SHOW "message25"
 } 2>&1 | tee -a $LOGFILE
-ROOTSIZE=`ls "$MAINDEST" -e1S | grep root | awk {'print $3'} `
-KERNELSIZE=`ls "$MAINDEST" -e1S | grep kernel | awk {'print $3'} `
+ROOTSIZE=`ls "$MAINDEST" -1S | grep root | awk {'print $3'} `
+KERNELSIZE=`ls "$MAINDEST" -1S | grep kernel | awk {'print $3'} `
 TOTALSIZE=$((($ROOTSIZE+$KERNELSIZE)/1024))
 SPEED=$(( $TOTALSIZE/$DIFF ))
 echo $SPEED > /usr/lib/enigma2/python/Plugins/Extensions/BackupSuite/speed.txt
