diff --git a/setup.py b/setup.py
index b76b74e..36c61df 100644
--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,7 @@ import os
 import platform
 import sys
 
-from setuptools import setup, Extension, Feature
+from setuptools import setup, Extension
 from setuptools.command.build_ext import build_ext
 from setuptools import find_packages
 
@@ -62,15 +62,7 @@ class optional_build_ext(build_ext):
 
 codeoptimization_c = os.path.join('src', 'zope', 'interface',
                                   '_zope_interface_coptimizations.c')
-codeoptimization = Feature(
-        "Optional code optimizations",
-        standard=True,
-        ext_modules=[
-            Extension(
-                "zope.interface._zope_interface_coptimizations",
-                [os.path.normcase(codeoptimization_c)]
-            )
-        ])
+
 py_impl = getattr(platform, 'python_implementation', lambda: None)
 is_pypy = py_impl() == 'PyPy'
 is_jython = 'java' in sys.platform
@@ -79,10 +71,7 @@ is_pure = 'PURE_PYTHON' in os.environ
 # Jython cannot build the C optimizations, while on PyPy they are
 # anti-optimizations (the C extension compatibility layer is known-slow,
 # and defeats JIT opportunities).
-if is_pypy or is_jython or is_pure:
-    features = {}
-else:
-    features = {'codeoptimization': codeoptimization}
+
 tests_require = ['zope.event']
 testing_extras = tests_require + ['nose', 'coverage']
 
@@ -148,6 +137,5 @@ setup(name='zope.interface',
           'test': tests_require,
           'testing': testing_extras,
       },
-      features=features,
       keywords=['interface', 'components', 'plugins'],
 )
