From 6465e96a344720ccc8b95b28497f2bf2019f430c Mon Sep 17 00:00:00 2001
From: fairbird <rrrr53@hotmail.com>
Date: Fri, 24 Apr 2020 15:19:10 +0300
Subject: [PATCH] Add_option_to_choose_style_of_VirtualKeyBoard

---

diff --git a/src/YouTubeSearch.py b/src/YouTubeSearch.py
index 9a005d6..1757619 100644
--- a/src/YouTubeSearch.py
+++ b/src/YouTubeSearch.py
@@ -5,6 +5,7 @@ from os import path as os_path
 from threading import Thread
 from json import loads
 
+import os
 from enigma import ePoint
 from Screens.ChoiceBox import ChoiceBox
 from Screens.Screen import Screen
@@ -17,6 +18,7 @@ from Components.Pixmap import Pixmap
 from Components.Sources.Boolean import Boolean
 from Components.Sources.List import List
 from Components.Sources.StaticText import StaticText
+from Tools.Directories import resolveFilename, SCOPE_PLUGINS
 
 from . import _, screenwidth
 from .compat import compat_quote
@@ -352,8 +354,20 @@ class YouTubeSearch(Screen, ConfigListScreen):
 
 	def openKeyboard(self):
 		self['config'].getCurrent()[1].help_window.instance.hide()
-		self.session.openWithCallback(self.keyBoardCallback, YouTubeVirtualKeyBoard,
-				text=self.searchValue.value)
+		self.VirtualKeyBoard = config.plugins.YouTube.VirtualKeyBoard.value
+		if self.VirtualKeyBoard == "YouTube":
+			if os.path.islink("/usr/lib/enigma2/python/Screens/VirtualKeyBoard.py") and os.path.exists(resolveFilename(SCOPE_PLUGINS, "SystemPlugins/NewVirtualKeyBoard")):
+				from Screens.VirtualKeyBoard import VirtualKeyBoard
+				self.session.openWithCallback(self.keyBoardCallback, VirtualKeyBoard, '')
+			else:
+				self.session.openWithCallback(self.keyBoardCallback, YouTubeVirtualKeyBoard,
+					text = self.searchValue.value)
+		elif self.VirtualKeyBoard == "Image":
+			from Screens.VirtualKeyBoard import VirtualKeyBoard
+			self.session.openWithCallback(self.keyBoardCallback, VirtualKeyBoard, '')
+		else:
+			self.session.openWithCallback(self.keyBoardCallback, YouTubeVirtualKeyBoard,
+				text = self.searchValue.value)
 
 	def keyBoardCallback(self, name):
 		config = self['config'].getCurrent()[1]
diff --git a/src/YouTubeUi.py b/src/YouTubeUi.py
index 07e2a36..f0fed29 100644
--- a/src/YouTubeUi.py
+++ b/src/YouTubeUi.py
@@ -76,6 +76,8 @@ config.plugins.YouTube.onMovieEof = ConfigSelection(default='related', choices=[
 config.plugins.YouTube.onMovieStop = ConfigSelection(default='related', choices=[
 	('related', _('Show related videos')),
 	('ask', _('Ask user')), ('quit', _('Return to list'))])
+config.plugins.YouTube.VirtualKeyBoard = ConfigSelection(default='YouTube', choices=[
+	('YouTube', _('YouTube VirtualKeyBoard')), ('Image', _('Image VirtualKeyBoard'))])
 config.plugins.YouTube.login = ConfigYesNo(default=False)
 config.plugins.YouTube.downloadDir = ConfigDirectory(default=resolveFilename(SCOPE_HDD))
 config.plugins.YouTube.useDashMP4 = ConfigYesNo(default=True)
@@ -1509,6 +1511,9 @@ class YouTubeSetup(ConfigListScreen, Screen):
 			self.list.append(getConfigListEntry(_('Merge downloaded files:'),
 				config.plugins.YouTube.mergeFiles,
 				_('FFmpeg will be used to merge downloaded DASH video and audio files.\nFFmpeg will be installed if necessary.')))
+		self.list.append(getConfigListEntry(_('Choose VirtualKeyBoard Style:'),
+			config.plugins.YouTube.VirtualKeyBoard,
+			_('You can choose what style of VirtualKeyBoard to use it.\nYouTube OR Image (VirtualKeyBoard).')))
 		for p in plugins.getPlugins(where=PluginDescriptor.WHERE_MENU):
 			# TRANSLATORS: Don't translate this! It is used as a variable, so it must be equal to the translation in the plugin!
 			if p.name == _('ServiceApp'):


