diff --git a/libopkg/opkg_conf.c b/libopkg/opkg_conf.c
index 06880a1..a0dd2f6 100644
--- a/libopkg/opkg_conf.c
+++ b/libopkg/opkg_conf.c
@@ -394,7 +394,7 @@ static int opkg_conf_parse_file(const char *filename,
                              regmatch[13].rm_eo - regmatch[13].rm_so);
         }
 
-        if (regmatch[15].rm_so != regmatch[15].rm_eo && strncmp(type, "dist", 4) != 0) {
+        if (regmatch[13].rm_so != regmatch[13].rm_eo && strncmp(type, "dist", 4) != 0) {
             opkg_msg(ERROR,
                      "%s:%d: Ignoring config line with trailing garbage: `%s'\n",
                      filename, line_num, line);
