diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 0ec774ae4..2d6b19fd3 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -22,6 +22,8 @@
 #include <config.h>
 #endif
 
+#define _GNU_SOURCE /* Enable mmap64() */
+
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
@@ -517,7 +519,7 @@ gst_v4l2_object_new (GstElement * element,
     v4l2object->dup = dup;
     v4l2object->ioctl = ioctl;
     v4l2object->read = read;
-    v4l2object->mmap = mmap;
+    v4l2object->mmap = mmap64;
     v4l2object->munmap = munmap;
   }
 
