diff --git a/external/ffmpeg/put_bits.h b/external/ffmpeg/put_bits.h
index 745691b..f7e52c4 100755
--- a/external/ffmpeg/put_bits.h
+++ b/external/ffmpeg/put_bits.h
@@ -202,7 +202,7 @@ static inline void put_sbits(PutBitContext *pb, int n, int32_t value)
 {
     av_assert2(n >= 0 && n <= 31);
 
-    put_bits(pb, n, av_mod_uintp2(value, n));
+    put_bits(pb, n, av_zero_extend(value, n));
 }
 
 /**
