diff --git a/IPTVPlayer/setup/iptvsetupimpl.py b/IPTVPlayer/setup/iptvsetupimpl.py
index 1cd7f132..0c6ff9b8 100644
--- a/IPTVPlayer/setup/iptvsetupimpl.py
+++ b/IPTVPlayer/setup/iptvsetupimpl.py
@@ -307,7 +307,7 @@ class IPTVSetupImpl:
     def setOpenSSLVersion(self, ret=None):
         printDBG('Check opennSSL version')
         self.setInfo(_("Detection of the OpenSSL version."), _("OpenSSL lib is needed by wget and rtmpdump utilities."))
-        for ver in ['.0.9.8', '.1.0.0', '.1.0.2']:
+        for ver in ['.0.9.8', '.1.0.0', '.1.0.2', '.1.1']:
             libsslExist = False
             libcryptoExist = False
             libSSLPath = ''
@@ -330,6 +330,8 @@ class IPTVSetupImpl:
         if libsslExist and libcryptoExist:
             self.openSSLVersion = ver
             self.libSSLPath = libSSLPath
+            if '.1.1' == ver:
+                self.getGstreamerVer()
             if '.1.0.2' == ver:
                 self.getOpenssl2Finished()
             elif '.0.9.8' == ver:
