diff --git a/configure.ac b/configure.ac
index 4e06609..1a28707 100644
--- a/configure.ac
+++ b/configure.ac
@@ -30,17 +26,6 @@ dnl check for tools
 AC_PROG_CC
 AC_PROG_LIBTOOL
 
-
-dnl decide on error flags
-AS_COMPILER_FLAG(-Wall, GST_WALL="yes", GST_WALL="no")
-
-if test "x$GST_WALL" = "xyes"; then
-   GST_ERROR="$GST_ERROR -Wall"
-   if test "x$GST_CVS" = "xyes"; then
-     AS_COMPILER_FLAG(-Werror,GST_ERROR="$GST_ERROR -Werror",GST_ERROR="$GST_ERROR")
-   fi
-fi
-
 dnl Check for pkgconfig first
 AC_CHECK_PROG(HAVE_PKGCONFIG, pkg-config, yes, no)
 
@@ -62,9 +47,6 @@ if test "x$HAVE_GST" = "xno"; then
   AC_MSG_ERROR(you need gstreamer development packages installed !)
 fi
 
-dnl append GST_ERROR cflags to GST_CFLAGS
-GST_CFLAGS="$GST_CFLAGS $GST_ERROR"
-
 dnl make GST_CFLAGS and GST_LIBS available
 AC_SUBST(GST_CFLAGS)
 AC_SUBST(GST_LIBS)
