diff --git a/main.c b/main.c
index 1c1fee7..f497502 100644
--- a/main.c
+++ b/main.c
@@ -1148,7 +1148,8 @@ void getvideo(unsigned char *video, int *xres, int *yres)
 			return;
 		}
 
-		int adr, adr2, ofs, ofs2, offset, pageoffset;
+		int adr2, ofs, ofs2, offset, pageoffset;
+		size_t adr;
 		int xtmp,xsub,ytmp,t2,dat1;
 
 		if (stb_type == BRCM73565 || stb_type == BRCM73625 || stb_type == BRCM7439DAGS || stb_type == BRCM7439 || stb_type == BRCM75845 || stb_type == BRCM72604) {
