diff --git a/IPTVPlayer/setup/iptvsetupimpl.py b/IPTVPlayer/setup/iptvsetupimpl.py
index 2d41065..68770ab 100644
--- a/IPTVPlayer/setup/iptvsetupimpl.py
+++ b/IPTVPlayer/setup/iptvsetupimpl.py
@@ -469,7 +469,7 @@ class IPTVSetupImpl:
                 if '.' == self.ffmpegVersion[-1]: self.ffmpegVersion = self.ffmpegVersion[:-1]
             except Exception: self.ffmpegVersion = ""
         else: self.ffmpegVersion = ""
-        self.wgetStep()
+        self.finish()
             
     ###################################################
     # STEP: WGET
