diff --git a/IPTVPlayer/setup/iptvsetupwidget.py b/IPTVPlayer/setup/iptvsetupwidget.py
index 6d5b541..35bb247 100644
--- a/IPTVPlayer/setup/iptvsetupwidget.py
+++ b/IPTVPlayer/setup/iptvsetupwidget.py
@@ -46,11 +46,6 @@ class IPTVSetupMainWidget(Screen):
         self["sub_title"]  = Label(_(" "))
         self["info_field"] = Label(_(" "))
         
-        self["actions"] = ActionMap(["SetupActions", "ColorActions"],
-            {
-                "cancel"  :  self.cancelPressed,
-                "ok"      :  self.startPressed,
-            }, -1)
         try:
             for idx in range(5):
                 spinnerName = "spinner"
@@ -85,7 +80,7 @@ class IPTVSetupMainWidget(Screen):
         printDBG("IPTVSetupMainWidget.onStart")
         self["sub_title"].setText(_("Information"))
         self["info_field"].setText(_("IPTVPlayer need some additional setup.\nSuch as downloading and installation additional binaries.\nPress OK to start."))
-        if self.autoStart: self.startPressed()
+        self.setupImpl.start()
         
     def cancelPressed(self):
         printDBG("IPTVSetupMainWidget.cancelPressed")
