diff --git a/AutoBouquetsMaker/lib/dvbreader.c b/AutoBouquetsMaker/lib/dvbreader.c
index 034dd7f..b89ac30 100644
--- a/AutoBouquetsMaker/lib/dvbreader.c
+++ b/AutoBouquetsMaker/lib/dvbreader.c
@@ -8,6 +8,20 @@
 
 #include <Python.h>
 
+/**
++ * DMX_SET_SOURCE and dmx_source enum removed on 4.14 kernel
++ * Check commit 13adefbe9e566c6db91579e4ce17f1e5193d6f2c
++**/
+enum dmx_source {
+	DMX_SOURCE_FRONT0 = 0,
+	DMX_SOURCE_FRONT1,
+	DMX_SOURCE_FRONT2,
+	DMX_SOURCE_FRONT3,
+};
+#define DMX_SET_SOURCE _IOW('o', 49, enum dmx_source)
+
+typedef enum dmx_source  dmx_source_t;
+
 PyObject *ss_open(PyObject *self, PyObject *args) {
 	int fd, pid;
 	const char *demuxer;
