diff --git a/gst/gstpad.c b/gst/gstpad.c
index ebc184c8c..c562e7f69 100644
--- a/gst/gstpad.c
+++ b/gst/gstpad.c
@@ -1373,7 +1373,7 @@ cleanup_hook (GstPad * pad, GHook * hook)
   GstPadProbeType type;
 
   GST_DEBUG_OBJECT (pad,
-      "cleaning up hook %" G_GUINT64_FORMAT " with flags %08x", hook->hook_id,
+      "cleaning up hook %lu" G_GUINT64_FORMAT " with flags %08x", hook->hook_id,
       hook->flags);
 
   if (!G_HOOK_IS_VALID (hook))
