diff --git a/plugins/elements/gstfilesink.c b/plugins/elements/gstfilesink.c
index de4cb542e..e8dbe699d 100644
--- a/plugins/elements/gstfilesink.c
+++ b/plugins/elements/gstfilesink.c
@@ -772,7 +772,7 @@ gst_file_sink_render_list (GstBaseSink * bsink, GstBufferList * buffer_list)
     gst_buffer_list_foreach (buffer_list, accumulate_size, &size);
 
     GST_DEBUG_OBJECT (sink,
-        "Queueing buffer list of %u bytes (%u buffers) at offset %"
+        "Queueing buffer list of %u bytes (%u buffers) at offset %llu"
         G_GSIZE_FORMAT, size, num_buffers,
         sink->current_pos + sink->current_buffer_size);
 
@@ -827,7 +827,7 @@ gst_file_sink_render (GstBaseSink * sink, GstBuffer * buffer)
           gst_buffer_get_size (buffer));
   } else if (n_mem > 0) {
     GST_DEBUG_OBJECT (filesink,
-        "Queueing buffer of %" G_GSIZE_FORMAT " bytes at offset %"
+        "Queueing buffer of %u" G_GSIZE_FORMAT " bytes at offset %llu"
         G_GSIZE_FORMAT, gst_buffer_get_size (buffer),
         filesink->current_pos + filesink->current_buffer_size);
 
